src/zkl.rs

```diff
@@ -0,0 +1,107 @@
+pub struct Parser {}
+
+impl Parser {
+    pub fn parse(proof_spec: &ProofSpec, witness: &Witness) -> Result<Proof, ParseError> {
+        unimplemented!();
+    }
+}
+
+pub struct ProofSpec {}
+
+pub struct ProofSpecBuilder {}
+
+impl ProofSpecBuilder {
+    pub fn new() -> ProofSpecBuilder {
+        ProofSpecBuilder{}
+    }
+}
+
+pub struct Witness {}
+
+pub struct WitnessBuilder {}
+
+impl WitnessBuilder {
+    pub fn new() -> WitnessBuilder {
+        WitnessBuilder {}
+    }
+}
+
+pub struct Proof {}
+
+impl Proof {
+    pub fn verify(&self, proof_spec: &ProofSpec) -> Result<bool, ParseError> {
```

Comment:
It shouldn't be a `ParseError`. Maybe lets make a generic error and work with that for now